Build with GNU Make fails if LDFLAGS is defined in command-line

In our project the root Makefile calls rebar which then builds exml as a dependency (it is about rebar2).

Tricky thing is that by invoking gmake as `gmake LDFLAGS="our_flags"` the knowledge of that command-line argument propagates through rebar to `gmake -C c_src` in `exml/rebar.config` directly to `exml/c_src/Makefile` 

At this moment gmake uses the overriding rules described in https://www.gnu.org/software/make/manual/html_node/Overriding.html#Overriding.

Particularly:

> An argument that contains ‘=’ specifies the value of a variable: ‘v=x’ sets the value of the variable v to x. If you specify a value in this way, all ordinary assignments of the same variable in the makefile are ignored; we say they have been overridden by the command line argument.


As a consequence the line `LDFLAGS += -shared` doesn't work and the build fails with error:

```
gmake[1]: Entering directory '/data/buildbot/slave/freebsd11/d41d8cd9-pr-bugtrace-new-exml-fedc8fba226eeb8031754354e0140e65/deps/exml/c_src'
c++ -I/usr/local/include -std=c++11 -Wall -fPIC -O3 -I /data/erlang/otp_18.3_hacked/lib/erlang/erts-7.3/include/ -I /data/erlang/otp_18.3_hacked/lib/erlang/lib/erl_interface-3.8.2/include  -c -o exml.o exml.cpp
c++ exml.o -L/usr/local/lib -L /data/erlang/otp_18.3_hacked/lib/erlang/lib/erl_interface-3.8.2/lib -lerl_interface -lei -o /data/buildbot/slave/freebsd11/d41d8cd9-pr-bugtrace-new-exml-fedc8fba226eeb8031754354e0140e65/deps/exml/c_src/../priv/exml_nif.so
/usr/lib/crt1.o: In function `_start':
/usr/src/lib/csu/amd64/crt1.c:(.text+0x17b): undefined reference to `main'
exml.o: In function `load(enif_environment_t*, void**, unsigned long)':
exml.cpp:(.text+0x2d): undefined reference to `enif_system_info'
exml.cpp:(.text+0x187): undefined reference to `enif_open_resource_type'
exml.cpp:(.text+0x193): undefined reference to `enif_alloc_env'
exml.cpp:(.text+0x1a9): undefined reference to `enif_make_atom'
exml.cpp:(.text+0x1c3): undefined reference to `enif_make_atom'
exml.cpp:(.text+0x1dd): undefined reference to `enif_make_atom'
exml.cpp:(.text+0x1f7): undefined reference to `enif_make_atom'
And so on.
```

The fix is easy. 
Remove:
`LDFLAGS += -shared`

And put -shared flag to the:

```
- $(link_verbose) $(CXX) $^ $(LDFLAGS) $(LDLIBS) -o $@
+ $(link_verbose) $(CXX) $^ $(LDFLAGS) -shared $(LDLIBS) -o $@
```
since you definitely know at that moment that you are going to build a shared library regardless of LDFLAGS.

Not sure that extra PR worth here :)

BTW: platform - FreeBSD.

--- a/c_src/Makefile
+++ b/c_src/Makefile
@@ -34,7 +34,6 @@ endif
 
 CXXFLAGS += -std=c++11 -Wall -fPIC -O3 -I $(ERTS_INCLUDE_DIR) -I $(ERL_INTERFACE_INCLUDE_DIR)
 LDLIBS += -L $(ERL_INTERFACE_LIB_DIR) -lerl_interface -lei
-LDFLAGS += -shared
 
 # Verbosity.
 
@@ -54,7 +53,7 @@ COMPILE_CPP = $(cpp_verbose) $(CXX) $(CXXFLAGS) $(CPPFLAGS) -c
 
 $(EXML_NIF): exml.o
 	@mkdir -p $(BASEDIR)/priv/
-	$(link_verbose) $(CXX) $^ $(LDFLAGS) $(LDLIBS) -o $@
+	$(link_verbose) $(CXX) $^ $(LDFLAGS) -shared $(LDLIBS) -o $@
 
 %.o: %.cpp
 	$(COMPILE_CPP) $(OUTPUT_OPTION) $<

